src/averageAges.js

```diff
@@ -55,10 +78,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter((person, _, arr) => {
+    const hasMother = arr.find(mother => mother.name === person.mother);
+
+    return hasMother && (onlyWithSon ? person.sex === 'm' : true);
+  });
+
+  const childrensTotalAge = children.reduce((acc, child,) => {
+    const mother = people.find(person => person.name === child.mother);
+
+    if (mother !== child.born) {
+      return acc + child.born - mother.born;
+    }
+  }, 0);
```

Comment:
It's almost correct here. But you need to return the `acc` outside of the `if` statement. Also, you have a syntax error in the `reduce` method parameters.